### PR TITLE
Add animations driven by audio

### PR DESCRIPTION
**Pull request by blopker**
This is a WIP for now. Comments welcome!

### CODE REVIEW

File: lib/app/circle/components/circle_live_totem_participant.dart [hunk 1]
```diff
@@ -44,42 +49,67 @@ class _CircleLiveTotemParticipantState
...
-            height: 126,
-            child: participant.hasImage
-                ? CachedNetworkImage(
-                    imageUrl: participant.sessionImage!, fit: BoxFit.cover)
-                : Container(),
-          ),
-        ),
-      ),
-    );
+  Widget participant(BuildContext context, SessionParticipant participant,
```
Comment (blopker) [inline lib/app/circle/components/circle_live_totem_participant.dart:60]: Not totally happy with this one yet. Going to tweak a bit more. Suggestions?

File: lib/app/circle/components/circle_participant.dart [hunk 2]
```diff
@@ -23,6 +23,7 @@ class CircleParticipant extends StatelessWidget {
     final textStyles = themeData.textTheme;
     final themeColors = themeData.themeColors;
     return ClipRRect(
+        clipBehavior: Clip.antiAliasWithSaveLayer,
```
Comment (blopker) [inline lib/app/circle/components/circle_participant.dart:26]: This fixes a small artifact from clipping. Hopefully won't cause performance issues.

File: lib/services/communication_provider.dart [hunk 3]
```diff
@@ -25,4 +25,5 @@ abstract class CommunicationProvider extends ChangeNotifier {
   Future<bool> receiveActiveSessionTotem({required String sessionUserId});
   Future<bool> passActiveSessionTotem({required String sessionUserId});
   Future<bool> doneActiveSessionTotem({required String sessionUserId});
+  Stream<CommunicationAudioVolumeIndication> get audioIndicatorStream;
```
Comment (blopker) [inline lib/services/communication_provider.dart:28]: Should this go somewhere else?

Comment (blopker) [reply lib/services/communication_provider.dart:28]: Maybe ActiveSession?

**Review by schalky — APPROVED** [review]
looks good

**Pull request merged**